src/workflow/ArcanistPatchWorkflow.php

```diff
@@ -1123,4 +1124,22 @@ private function buildDependencyGraph(ArcanistBundle $bundle) {
     return $graph;
   }
 
+  private function pullBaseTagFromStagingArea($id){
+    list($success, $message, $staging, $staging_uri) = $this->validateStagingSetup();
+    if (!$success) {
+      return $message;
+    }
+    $prefix = idx($staging, 'prefix', 'phabricator');
+    $base_tag = "{$prefix}/base/{$id}";
+    echo pht('Fetching base tag from staging remote')."\n";
+    $err = phutil_passthru(
+          'git fetch --tag -n %s %s',
```

Comment:
Works for repos with no staging remote. 

varadarb-C02SG7Q3G8WP:uci-engine varadarb$ ../../uber_arcanist/bin/arc patch --diff 3694579
 **SKIP STAGING  No staging area is configured for this repository.**
Created and checked out branch arcpatch-D1111695.
Checking patch README.md...
Applied patch README.md cleanly.
 OKAY  Successfully committed patch